R/rdeephaven/R/helper_functions.R

```diff
@@ -1,18 +1,86 @@
-.verify_string <- function(arg_name, string_candidate) {
-  if (class(string_candidate)[[1]] != "character") {
-    stop(paste0("'", arg_name, "' must be passed as a single string. Got an object of class ", class(string_candidate)[[1]], " instead."))
-  } else if (length(string_candidate) != 1) {
-    stop(paste0("'", arg_name, "' must be passed as a single string. Got a character vector of length ", length(string_candidate), " instead."))
+first_class <- function(arg) {
+  return(class(arg)[[1]])
+}
+
+verify_type <- function(arg_name, candidate, required_type, message_type_name, is_scalar) {
+  if (first_class(candidate) == "list") {
```

Comment:
I moved the `if(is_scalar)` on line 20 up to the top, so we check for scalar first, and do not try to evaluate the lists if that fails. I think this is what you mean, not totally sure.

Reply:
I meant to change this line to something like:
```
if (!is_scalar && first_class(candidate) == "list") {
```
so that the grammar for all of your checks is correct for scalar vs vector as well as having all of the checks be correct.  That is the fundamental thing I was getting at.  Correctness of check and grammar that makes sense to the user.  e.g. if the check expects a scalar, why can our message talk about vectors and plural?